TTS/trainer.py

```diff
@@ -764,13 +765,16 @@ def test_run(self) -> None:
         """Run test and log the results. Test run must be defined by the model.
         Model must return figures and audios to be logged by the Tensorboard."""
         if hasattr(self.model, "test_run"):
-            if hasattr(self.eval_loader.load_test_samples):
-                samples = self.eval_loader.load_test_samples(1)
-                figures, audios = self.model.test_run(samples)
+            if isinstance(self.model, Wavegrad):
+                return None  # TODO: Fix inference on WaveGrad
+            if hasattr(self.eval_loader.dataset, "load_test_samples"):
+                samples = self.eval_loader.dataset.load_test_samples(1)
+                figures, audios = self.model.test_run(self.ap, samples, None, self.use_cuda)
```

Comment:
IMO we don't need to pass `use_cuda` but use the model's device instead

Reply:
Oh good, so a slef.device would work in the test_run ? I'll do the change

Reply:
self.device would not but `next(self.parameters().device)`

Reply:
next(self.parameters().device) == "cuda",
AttributeError: 'generator' object has no attribute 'device'
Did you mean `next(self.parameters()).device`?